modules/internal/ChapelBase.chpl

```diff
@@ -1286,23 +1306,29 @@ module ChapelBase {
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs:real(?w), rhs:real(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0.0 then
+        halt("Attempt to divide by zero");
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs, rhs) {
     lhs = lhs / rhs;
   }
 
   inline proc %=(ref lhs:int(?w), rhs:int(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0:int(w) then
```

Comment:
I'd advocate for being brave / breaking superstitions for things that it seems the language should do.  It's a good way to find real problems when it doesn't do what you think it should.